specification/resources/resource-manager/Microsoft.Resources/preview/2022-06-01-preview/validateResources.json

```diff
@@ -0,0 +1,403 @@
+    "description": "Provides operations for working with resource validation."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/providers/Microsoft.Resources/validateResources": {
+      "post": {
+        "tags": [
+          "ValidateResources"
+        ],
+        "operationId": "Deployments_ValidateResourcesAtTenantScope",
+        "description": "Validates whether the specified resources are syntactically correct and will be accepted by Azure Resource Manager.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ResourceValidatorRequest"
+            },
+            "description": "Parameters to validate."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Returns the validation result.",
+            "schema": {
+              "$ref": "#/definitions/ResourceValidatorResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Resource validation": {
+            "$ref": "./examples/PostValidateResources.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "CloudError": {
+      "x-ms-external": true,
+      "properties": {
+        "error": {
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/ErrorResponse"
+        }
+      },
+      "description": "An error response for a resource management request."
+    },
+    "ExtendedLocation": {
+      "properties": {
+        "type": {
+          "type": "string",
+          "enum": [
+            "EdgeZone"
+          ],
+          "x-ms-enum": {
+            "name": "ExtendedLocationType",
+            "modelAsString": true
+          },
+          "description": "The extended location type."
+        },
+        "name": {
+          "type": "string",
+          "description": "The extended location name."
+        }
+      },
+      "description": "Resource extended location."
+    },
+    "GenericResource": {
+      "properties": {
```

Comment:
>properties [](http://example.com/codeflow?start=7&length=10)

how about tags, location? why not use the the common definition for ResourceModelWithAllowedPropertySet instead?

[https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/types.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/types.json)

Reply:
Good point, I did a copy & paste of the previous api spec, my assumption is that when the initial spec got created we didn't have common types.  For now I'll leave it as is with a note that I might probably simplify the spec in later versions.